### ISSUE TITLE
Crash after SymPy fails to parse variable named `is` in mod file

### ISSUE THREAD

**Issue by kbvw** [BlueBrain/nmodl#1258]
When running NMODL on a mod file (`kv4_ch.mod`, found in https://github.com/Hjorthmedh/BasalGangliaData/tree/main/data/neurons/mechanisms) containing a variable named `is`, SymPy throws a `SyntaxError` that later results in NMODL crashing with seemingly unrelated errors.

```
(.venv_nmodl) vanwalst@bbpv1 ~/p/snudda> nmodl BasalGangliaData/data/neurons/mechanisms/kv4_ch.mod
INFO : Using nmodl-nightly Package (Developer Version)
[NMODL] [warning] :: SympySolverVisitor :: solve_non_lin_system python exception: Sympify of expression 'could not parse 'is'' failed, because of exception being raised:
SyntaxError: invalid syntax (<string>, line 1)
[NMODL] [warning] :: SympySolverVisitor :: solve_non_lin_system python exception: Sympify of expression 'could not parse 'is'' failed, because of exception being raised:
SyntaxError: invalid syntax (<string>, line 1)
[NMODL] [error] :: NeuronSolveVisitor :: solver method 'sparse' not supported
[NMODL] [error] :: NeuronSolveVisitor :: solver method 'sparse' not supported
[NMODL] [error] :: NeuronSolveVisitor :: solver method 'sparse' not supported
[NMODL] [error] :: NeuronSolveVisitor :: solver method 'sparse' not supported
[NMODL] [error] :: NeuronSolveVisitor :: solver method 'sparse' not supported
[NMODL] [error] :: NeuronSolveVisitor :: solver method 'sparse' not supported
[NMODL] [error] :: NeuronSolveVisitor :: solver method 'sparse' not supported
[NMODL] [error] :: NeuronSolveVisitor :: solver method 'sparse' not supported
[NMODL] [error] :: NeuronSolveVisitor :: solver method 'sparse' not supported
[NMODL] [error] :: NeuronSolveVisitor :: solver method 'sparse' not supported
[NMODL] [error] :: NeuronSolveVisitor :: solver method 'sparse' not supported
[NMODL] [error] :: NeuronSolveVisitor :: solver method 'sparse' not supported
[NMODL] [error] :: NeuronSolveVisitor :: solver method 'sparse' not supported
[NMODL] [error] :: NeuronSolveVisitor :: solver method 'sparse' not supported
[NMODL] [error] :: NeuronSolveVisitor :: solver method 'sparse' not supported
[NMODL] [error] :: NeuronSolveVisitor :: solver method 'sparse' not supported
[NMODL] [error] :: NeuronSolveVisitor :: solver method 'sparse' not supported
[NMODL] [error] :: NeuronSolveVisitor :: solver method 'sparse' not supported
[NMODL] [error] :: NeuronSolveVisitor :: solver method 'sparse' not supported
[NMODL] [error] :: NeuronSolveVisitor :: solver method 'sparse' not supported
[NMODL] [error] :: NeuronSolveVisitor :: solver method 'sparse' not supported
[NMODL] [error] :: NeuronSolveVisitor :: solver method 'sparse' not supported
terminate called after throwing an instance of 'std::runtime_error'
  what():  PRIME encountered during code generation, ODEs not solved?
fish: Job 1, 'nmodl BasalGangliaData/data/neu…' terminated by signal SIGABRT (Abort)
```

After renaming the variable, NMODL runs without error. Tested on nightly 0.6.152 and on 0.5.

**Comment by kbvw** [comment]
NMODL continuing to run after the SymPy error will be addressed by #1257.

However, it is not clear to me if a variable named `is` should or should not be allowed in a mod file?

**Comment by 1uc** [comment]
We've noticed the same yesterday. NEURON and it's MOD file converter allow it. Plus, it happens only when we send some string (containing `is`) to SymPy that things go wrong.

If you need it to works with NMODL soon, you're probably best off changing the variable name to something more descriptive or anything not `is`, like `i_s` or `is_`.
